src/Swarm/Game/State.hs

```diff
@@ -248,6 +254,15 @@ robotMap :: Lens' GameState (IntMap Robot)
 --   happen.
 robotsByLocation :: Lens' GameState (Map (V2 Int64) IntSet)
 
+-- | The list of entities that have been discovered.
+allDiscoveredEntities :: Lens' GameState Inventory
+
+-- | The list of devices that have been installed once.
+allInstalledDevices :: Lens' GameState Inventory
+
+-- | The list of available recipes, with the position of the last known item.
+availableRecipes :: Lens' GameState (Int, [Recipe Entity])
```

Comment:
I don't think I understand what the `Int` is for.

Reply:
Oh, wait, I get it, it's to keep track of which available recipes are "read" and which are "unread", right?

Reply:
Yes, and I'd like to add a separator so that the new recipes are separated from the known one.